### ISSUE TITLE
antsApplyTransforms and all-zero pixel value issue

### ISSUE THREAD

**Issue by s-j-choi** [ANTsX/ANTs#546]
Dear experts,

I am very new to ANTs tool.

I was able to run a few scripts using "antsRegistration" and acquired results that I expected. Then, I was trying to use "antsApplyTransforms" in a script as given below:
----------------------------------------------------------------------------------------------------------------------
###########################################
#
# Presets for ANTs
#
###########################################
applyXFM=antsApplyTransforms
dim=3
fixed=SS_T1map.nii.gz
moving=R2star-BETed.nii.gz
moved=GRE-2-T1map_s-Warped_R2star.nii.gz
affine=GRE-2-T1map_s-0GenericAffine.mat
warp=GRE-2-T1map_s-1Warp.nii.gz

scancode=7TMS_008_08OCT2014
	cd $dir_work/$scancode/
	if [ ! -f $scancode-$moved ]; then
		time $dir_ANTs/$applyXFM -d $dim \
								-e 0 \
								-i $scancode-$moving \
								-o $scancode-$moved \
								-r $scancode-$fixed \
								-n BSpline[ 3 ] \
								-t $scancode-$warp \
								-t $scancode-$affine \
								--float 0
		fslview $scancode-$fixed $scancode-$moved -b 0, 300
	else
		echo "$scancode-GRE-2-T1map_s-Warped_R2star.nii.gz alredy exists."
		fslview $scancode-$fixed $scancode-$moved -b 0, 300
	fi
---------------------------------------------------------------------------------------------------------------------- 

When I run the script, it seemed to be working and when it was loaded into fslview, all pixel-values of the warped image turned out to be zero.

I checked $moving and $fixed. All input images were good.

I wonder if there are any points that I missed.

If you could take a look the attached simple script and provide me with any suggestions or opinions, I would appreciate it.
If I have missed any important details for you to figure out what my issue, please let me know. I will provide further information as requested.

I was running the script "Neurodebian" environment.

**Comment by s-j-choi** [comment]
Thanks for your reply. 
However, one thing looked odd to me was that the file size of the warped image appeared to be not right. It was less than 1 MB, which may reflect all pixels of zero values as in a mask of all zero values. Total applying transformation time took very short (< 1 min).

I will check if things look different with different viewers.

**Comment by s-j-choi** [comment]
By the way, results from antsRegistration in my other script did well and I was able to see the results using FSLVIEW without any issues. However, somehow, I don't understand that I cannot see the results from antsApplyTransforms because saving results (with ants environment) should be using the same software routines I guess.

**Comment by cookpa** [comment]
FSL might interpret the header differently, but it should still display the same volume of voxels. If the image is all zero, that suggests that `antsApplyTransforms` sampled points that were all background (or outside the moving image domain entirely).

From what I can make out, you registered a moving to a fixed image, and then are applying the transforms to another moving image. This will only work if the physical space of the moving image is the same for the call to `antsRegistration` and `antsApplyTransforms`.

**Comment by s-j-choi** [comment]
I am grateful to both of you for the suggestions and opinions. 

I fixed the problem. There were two issues folded. One issue was the precision of the data. Another was the header information. My QSM worked with the script without any issues. Hence I fixed the header info of the R2* using the QSM header info, since R2* and QSM are originated from an identical space. My R2* map was calculated using in-house Matlab code and the saved results had incorrect header info. After clearing this issue (using fslcpgeom), some warnings from FSLVIEW disappeared. However, the images still had all zero pixel values. Checked the data and changes its data type from FLOAT64 to FLOAT32  (using fslmaths). Then, all problems cleared out. 

To confirm the double folded issue, I checked the process by only changing the data type (FLOAT64 to FLOAT32) without fixing the header information. I got all zero pixel values. In conclusion, either correcting header info (fslcpgeom) or changing data type (FLOAT64 to FLOAT32) solely did not clear the issue.

The issue has been cleared. However, I am still don't understand why the data type (FLOAT32 or FLOAT64) does affect the pixel values after applying combined transforms.
